src/windows.rs

```diff
@@ -0,0 +1,79 @@
+use core::ptr;
+use core::ffi::{c_void};
+use once_cell::sync::Lazy;
+use Allocator;
+
+pub struct System {
+    _priv: (),
+}
+
+
+impl System {
+    pub const fn new() -> System { System { _priv: () } }
+}
+
+unsafe impl Allocator for System {
+    fn alloc(&self, size: usize) -> (*mut u8, usize, u32) {
+            let addr = unsafe {
+                windows::Win32::System::Memory::VirtualAlloc(
+                    ptr::null_mut(),
+                    size,
+                    windows::Win32::System::Memory::MEM_RESERVE | windows::Win32::System::Memory::MEM_COMMIT,
+                    windows::Win32::System::Memory::PAGE_READWRITE,
+                )
+            };
+
+            if addr == ptr::null_mut() {
+                (ptr::null_mut(), 0, 0)
+            } else {
+                (addr as *mut u8, size, 0)
+            }
+    }
+
+    fn remap(&self, ptr: *mut u8, oldsize: usize, newsize: usize, can_move: bool) -> *mut u8 {
+        ptr::null_mut()
+    }
+
+    fn free_part(&self, ptr: *mut u8, oldsize: usize, newsize: usize) -> bool { false }
+
+    fn free(&self, ptr: *mut u8, size: usize) -> bool {
+        unsafe {
+            windows::Win32::System::Memory::VirtualFree(
+                ptr as *mut c_void,
+                0,
+                windows::Win32::System::Memory::MEM_RELEASE).0 != 0
+        }
+    }
+
+    fn can_release_part(&self, _flags: u32) -> bool { true }
+
+    fn allocates_zeros(&self) -> bool { true }
+
+    fn page_size(&self) -> usize { 4096 }
+}
+
+#[cfg(feature = "global")]
+static LOCK: Lazy<windows::Win32::Foundation::HANDLE> = unsafe {
+    Lazy::new(|| {
```

Comment:
Given how low-level this crate is I think it would be best to not use an external crate to perform this initialization and instead internally use locking/synchronization as necessary.

Reply:
The thing is that I actually tried to initialize the lock without relying on external crate, but rust compiler didn't allow me to proceed displaying the following error message: `calls in statics are limited to constant functions, tuple structs and tuple variants`. So, I was advised to lazy-initialize the lock with the help of this crate. I'd be glad if you suggest an alternative

Reply:
Sorry I don't know how best to fix that, but regardless I don't think that using an external crate is appropriate in this case for a lock around the allocator.

Reply:
I think the best way to do this is to set the lock to `Handle(0)` at first. Then, in the lock functions, check if its equal to that and create the mutex if it is.